### PR TITLE
Initial guess, automatic scaling, multi-DOF power limit 

### LINKED ISSUE #39: Feature request: use previous solution for initial guess
**Feature description.**
When running design optimization studies, it is often the case that you will be calling `WecOptTool.solve` for a number of different design configurations (e.g., slightly different geometries). If the optimization algorithm you're using executes these calls serially, you can use the output of one `wec.solve` to set the initial guess (`x_wec_0` and `x_opt_0`) for the decision vector used in the subsequent `wec.solve` call. While the solution of the pseudo-spectral problem will change based on the `wec`, if the `wec`'s are similar, as is often the case, the initial guess will be close to the optimal solution of `x` and thus reduce computational time.

**Describe the solution you'd like**
Probably no need to alter the source for this; just show how this can be done via examples

### LINKED ISSUE #69: Power Limit in multiple DOFs
We need to verify that the power limit function works for multiple DOFs.

### LINKED ISSUE #74: Better scaling and/or initial guess
Do one or both of the following:

 - Run w/o inequality constraints
 - find the analytical solution (v = Fe/2/Ri)

From these, you can do one or both of the following:

- set `x0`
- set scaling

related to #4 #48

### PR DESCRIPTION
**Pull request by ryancoe**
## Description
 - [x] Fixes #74
 - [x] Fixes #69

## Checklist for PR
- [x] Authors read the [contribution guidelines](https://github.com/SNL-WaterPower/WecOptTool/blob/main/.github/CONTRIBUTING.md)
- [x] The pull request is from an issue branch (not main) on your fork to the [main branch in WecOptTool](https://github.com/SNL-WaterPower/WecOptTool). 
- [x] The authors have given the admins edit access
- [x] All changes adhere to the style guide including PEP8, Docstrings, and Type Hints.
- [x] Added note in [Changelog](https://github.com/SNL-WaterPower/WecOptTool/blob/main/CHANGES.md)
- [x] Modified the documentation if applicable
- [x] Modified or added a new test
- [x] All tests pass
- [x] Documentation builds
- [x] [Reference or close any relevant issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

### CODE REVIEW

File: tests/test_wecopttool.py [hunk 1]
```diff
@@ -137,6 +137,36 @@ def test_solve(wec, regular_wave, pto):
...
+        nits.append(res['nit'])
+        x_wecs.append(x_wec)
+
+    assert nits[0] > nits[1]
+    assert pytest.approx(x_wecs[1],1e0) == x_wec_0
+    
+    
+def test_complex_to_real_amplitudes(wec, regular_wave):
+    x_wec = wec.initial_x_wec_guess(regular_wave)
+    fd_wec = wot.real_to_complex_amplitudes(np.atleast_2d(x_wec))
```
Comment (ryancoe) [inline tests/test_wecopttool.py:164]: One consistency thing we should look at is the number of dimensions for `x_wec`. It seems that it is generally treated as 1D, but needs to be 2D for `real_to_complex_amplitudes`

File: wecopttool/core.py [hunk 2]
```diff
@@ -1106,6 +1203,12 @@ def power_limit(excitation: npt.ArrayLike, impedance: npt.ArrayLike
     """
 
     power_limit = -1*np.sum(np.abs(excitation)**2 / (8*np.real(impedance)))
+    
+    pls = np.concatenate([np.linalg.lstsq(8*impedance[w_ind, :, :].real,
```
Comment (ryancoe) [inline wecopttool/core.py:1199]: Add a test for power limits in multiple DOFs

File: wecopttool/core.py [hunk 3]
```diff
@@ -608,6 +620,25 @@ def _get_state_scale(self,
...
+
+        Returns
+        -------
+        x_wec_0
+            Initial guess for `x_wec`
+        """
+        pos_opt = self.optimal_position(waves)
+        pos_opt_zero_mean = np.concatenate([np.zeros((1, self.ndof)), pos_opt])
+        x_wec_0 = complex_to_real_amplitudes(pos_opt_zero_mean)
+        return x_wec_0.squeeze()
```
Comment (ryancoe) [inline wecopttool/core.py:640]: this relates to the question about the number of dimensions for `x_wec`

File: tests/test_wecopttool.py [hunk 4]
```diff
@@ -529,7 +578,34 @@ def surge_heave_wavebot():
...
+                                                      optim_options={
+                                                          'maxiter': 250, 
+                                                          'ftol': 1e-8},
+                                                      scale_x_wec=1e2,
+                                                      scale_x_opt=1e-2,
+                                                      scale_obj=1,
+                                                      )
+
+    plim = power_limit(fdom['excitation_force'], surge_heave_wavebot.hydro.Zi)
+    assert pytest.approx(obj, 1e-1) == plim
```
Comment (ryancoe) [inline tests/test_wecopttool.py:605]: Would've liked for this tolerance to be tighter...

File: tests/test_wecopttool.py (hunk 1 above)
Comment (cmichelenstrofer) [reply tests/test_wecopttool.py:164]: Should be 1D. Might need to modify `real_to_complex_amplitudes` to include an `np.atleast_2D`

**Review by cmichelenstrofer — APPROVED** [review]
Need to resolve conflicts with the other PR just merged into main.

File: tests/test_wecopttool.py (hunk 1 above)
Comment (ryancoe) [reply tests/test_wecopttool.py]: resolved w. 86c5580cd2f153beb24b7e08131bc949470803b3

File: wecopttool/core.py (hunk 2 above)
Comment (ryancoe) [reply wecopttool/core.py:1288]: resolved w. 8660b0da84cbb929b4f6f4645e30ec6ab3690c92

File: wecopttool/core.py (hunk 3 above)
Comment (ryancoe) [reply wecopttool/core.py]: resolved w. 61a69122d7e571722693187399b92b853dd1cb18

File: wecopttool/core.py [hunk 5]
```diff
@@ -1088,14 +1134,57 @@ def run_bem(fb: cpy.FloatingBody, freq: Iterable[float] = [np.infty],
...
+    impedance: np.ndarray
+        Complex impedance matrix. Shape: ``nfreq`` x ``ndof`` x ``ndof``
+
+    Returns
+    -------
+    opt_vel
+        Optimal velocity for power absorption.
+    """
+    opt_vel = np.concatenate([np.linalg.lstsq(2*impedance[w_ind, :, :].real,
+                                              excitation[w_ind+1, :])[0]
```
[the remark this answers is not in the document]
Comment (ryancoe) [reply wecopttool/core.py:1245]: noted in #86, closing here

**Pull request merged**